client/lib/cart/store/index.js

```diff
@@ -241,6 +241,7 @@ function createListener( store, selector, callback ) {
 
 // Subscribe to the Redux store to get updates about the selected site
 getReduxStore().then( store => {
-	CartStore.setSelectedSiteId( getSelectedSiteId( store.getState() ) );
+	const selectedSiteId = getSelectedSiteId( store.getState() );
+	selectedSiteId && CartStore.setSelectedSiteId( selectedSiteId );
```

Comment:
The reason I did this change was because while logged out, opening the Quick Start offer page showed the following error in the console:

![](https://user-images.githubusercontent.com/1269602/59622436-2fafb900-914f-11e9-8393-5fcd6f8f29c8.png)

> The original code set the cart site ID to null if some "All Sites" view is active. That disables the cart synchronization for that site.

In the following line, we still subscribe to the the redux store so wouldn't any null values still get set? Nevertheless, in [e1f863c](https://github.com/Automattic/wp-calypso/pull/33749/commits/e1f863c94e6e7812e531161b7dfcf2068dbdff82), I have removed this change. I have introduced a change wherein `setSelectedSiteId` proceeds only if the user is logged in, so that while logged out we don't poll the server for syncing the cart. Would this work, or is there something I am overlooking?